### PR TITLE
Test + fix for get-commit-test.js to return correct path in branchUrl

### PR DESCRIPTION
**Pull request by jywarren**
This test highlights an issue which I have written a fix for; adding the fix as a next step. Thanks!

`(.+)` is changed to `(\w+)` and the slash is moved inside the "blob" bit. I tested and it correctly addresses the issue in #189. It had been grabbing all of `blob/branch-name/path/to/file.rb` and replacing, instead of just `blob/branch-name`.

### CODE REVIEW

**Comment by jywarren** [comment]
Great! This failed, so #190 fixes it -- #191 includes both this test and the fix.

**Comment by jywarren** [comment]
@gr2m - here's the PR with both the failing test and the commit that fixes it, both with CI test results showing. Does this work for you? Thanks!

**Comment by jywarren** [comment]
Hi, @gr2m -- will this work for you? Thanks!

**Comment by jywarren** [comment]
Hmm, additional commits pushed this out of sync...

**Comment by jywarren** [comment]
It's weird, I remove `docs` from this line and then it switches to requiring `docs`... back and forth?

```
test/unit/get-commit-test.js .......................... 8/9
  get commit request succeeds
  not ok should be equal
    +++ found
    --- wanted
    -https://github.com/Techforchange/first-timers-test/blob/defaultBranch/docs/README.md
    +https://github.com/Techforchange/first-timers-test/blob/defaultBranch/README.md
```

**Comment by jywarren** [comment]
Any help appreciated! This is a small but important barrier we're trying to fix for our newcomers. Thank you!!! 😄

**Comment by jywarren** [comment]
As of 14b9efc it definitely worked!

**Comment by gr2m** [comment]
hey @jywarren I’m so sorry to not get back to you sooner, the PR and issue somehow managed to slipped out of my todo list! I’m looking into it now

File: test/unit/get-commit-test.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ test('get commit request succeeds', t => {
       t.is(state.commit.message, 'message')
       t.is(state.commit.filename, 'filename')
       t.is(state.commit.patch, 'patch')
-      t.is(state.commit.branchUrl, 'https://github.com/Techforchange/first-timers-test/blob/defaultBranch/README.md')
+      t.is(state.commit.branchUrl, 'https://github.com/Techforchange/first-timers-test/blob/defaultBranch/docs/README.md')
```
Comment (gr2m) [inline test/unit/get-commit-test.js]: you have to change line 28 to make it work

```diff
- blob_url: 'https://github.com/Techforchange/first-timers-test/blob/f00ecb1bbffe515500558568ae0b176d2a1defe8/README.md'
+ blob_url: 'https://github.com/Techforchange/first-timers-test/blob/f00ecb1bbffe515500558568ae0b176d2a1defe8/docs/README.md'
```

File: lib/get-commit.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ function getCommit (state) {
 
     .then(function (result) {
       const {filename, patch, blob_url: blobUrl} = result.data.files[0]
-      var branchUrl = blobUrl.replace(/(?:blob)(.+)(?=\/)/g, 'blob/' + state.repoDefaultBranch)
+      var branchUrl = blobUrl.replace(/(?:blob\/)(\w+)/g, 'blob/' + state.repoDefaultBranch)
```
Comment (gr2m) [inline lib/get-commit.js]: I think we I’d change the regex pattern

```diff
- /(?:blob\/)(\w+)/g
+ /\/blob\/[^/]+/
```

That should make it more clear. We don’t need the `g` modifier, either, it’s a single replace. 

The regex will cause problems in edge cases where a repository owner or repository name is "blob", but let’s create a follow up issue for that, it’s out of scope of this PR

**Comment by jywarren** [comment]
Ugh, that was an oversight! Thanks for the catch, let's see how this does. Sorry for the extra commits but it'll all be squashed anyways, right?

**Comment by jywarren** [comment]
OK, phew! This looks good again now. Will this work for squash-and-merge? Thanks @gr2m !!

**Comment by gr2m** [comment]
> Sorry for the extra commits but it'll all be squashed anyways, right?

Yes no problem at all :)

**Comment by jywarren** [comment]
Awesome! Do I need to take additional steps to see this deployed on our
install of first-timers-bot? Thanks so much!!!! That bot has been a really
awesome tool for us. Very grateful.

**Pull request merged**